python/lsst/ts/eas/eas_csc.py

```diff
@@ -90,6 +93,7 @@ def __init__(
         self.enabled_event = (
             asyncio.Event()
         )  # An event that is set when the CSC is enabled.
+        self.control_loop_lock = asyncio.Lock()
```

Comment:
Why do you need a lock?

Reply:
The lock prevents the control loop and shutdown routine from running at the same time.